core/foundation/inc/RtypesCore.h

```diff
@@ -34,91 +45,91 @@ class TRootIOCtor;
 #endif
 #ifdef R__B64    // Note: Long_t and ULong_t are currently not portable types
-typedef int            Seek_t;      //File pointer (int)
-typedef long           Long_t;      //Signed long integer 8 bytes (long)
-typedef unsigned long  ULong_t;     //Unsigned long integer 8 bytes (unsigned long)
+typedef int            Seek_t;      ///< File pointer (int). Size depends on architecture \deprecated Consider replacing with `int`.
+typedef long           Long_t;      ///< Signed long integer 8 bytes (long). Size depends on architecture \deprecated Consider replacing with `long`
+typedef unsigned long  ULong_t;     ///< Unsigned long integer 8 bytes (unsigned long). Size depends on architecture \deprecated Consider replacing with `unsigned long`
 #else
-typedef int            Seek_t;      //File pointer (int)
-typedef long           Long_t;      //Signed long integer 4 bytes (long)
-typedef unsigned long  ULong_t;     //Unsigned long integer 4 bytes (unsigned long)
+typedef int            Seek_t;      ///< File pointer (int). Size depends on architecture \deprecated Consider replacing with `int`.
+typedef long           Long_t;      ///< Signed long integer 4 bytes (long). Size depends on architecture \deprecated Consider replacing with `long`
+typedef unsigned long  ULong_t;     ///< Unsigned long integer 4 bytes (unsigned long). Size depends on architecture \deprecated Consider replacing with `unsigned long`
 #endif
-typedef float          Float_t;     //Float 4 bytes (float)
-typedef float          Float16_t;   //Float 4 bytes written with a truncated mantissa
-typedef double         Double_t;    //Double 8 bytes
-typedef double         Double32_t;  //Double 8 bytes in memory, written as a 4 bytes float
-typedef long double    LongDouble_t;//Long Double
-typedef char           Text_t;      //General string (char)
-typedef bool           Bool_t;      //Boolean (0=false, 1=true) (bool)
-typedef unsigned char  Byte_t;      //Byte (8 bits) (unsigned char)
-typedef short          Version_t;   //Class version identifier (short)
-typedef const char     Option_t;    //Option string (const char)
-typedef int            Ssiz_t;      //String size (int)
-typedef float          Real_t;      //TVector and TMatrix element type (float)
-typedef long long          Long64_t; //Portable signed long integer 8 bytes
-typedef unsigned long long ULong64_t;//Portable unsigned long integer 8 bytes
+typedef float          Float_t;     ///< Float 4 bytes (float) \deprecated Consider replacing with `float`.
+typedef float          Float16_t;   ///< Float 4 bytes in memory, written to disk as a 3 bytes (24-bits, not 16-bits) float with a truncated (up-to) 16-bit mantissa (12-bit by default in memory), and (7+1)-bits of exponent \deprecated Consider (non-equivalent) alternatives in `<cstdfloat>`. \warning Do not mix this with a half-float such as std::float16_t
```

Comment:
This is either too much or not enough details.  The actual size on disk depends on the customization.

Reply:
But isn't the default 3 bytes? Since you are streaming a UChar_t and a UShort_t ?